zos-utils/src/main/java/org/zowe/commons/attls/AttlsContextImpl.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.commons.attls;
+
+/**
+ * This class publish all AT-TLS information about the session. As input are two parameters:
+ * - id - id of filedescription to attach right session
+ * - alwaysLoadCertificate - define, if certificate should be fetch together with basic information. It can save one
+ * more call of IOCTL in case certificate is always required. Otherwise library fetch certificate only if you call
+ * method getCertificate.
+ * <p>
+ * This context read value lazy, it means that until you call any method, no data are fetched. In case of getting any
+ * information, library make call on first time to query or certificate and cache them in internal block of bytes
+ * (see {@link AttlsContext#ioctl}. All java object are then created just once time and store into cache properties.
+ * Next calling returns only those cached value. If you want to fetch new data, you should call method
+ * {@link AttlsContext#clean()}.
+ * <p>
+ * For fetching a certificate is needed to prepare memory before, its size is defined by
+ * {@link AttlsContext#BUFFER_CERTIFICATE_LENGTH}.
```

Comment:
```suggestion
 * {@link AttlsContextImpl#BUFFER_CERTIFICATE_LENGTH}.
```

Maybe also the links above should be rather linked to this implementation (in this scope)